cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaQuorumCheck.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.QuorumInfo;
+
+/**
+ * Determines whether the given controller can be rolled without affecting the quorum.
+ */
+class KafkaQuorumCheck {
+
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(KafkaQuorumCheck.class.getName());
+
+    private final Admin ac;
+
+    private final Reconciliation reconciliation;
+
+    private final Future<QuorumInfo> quorumInfoFuture;
+
+    KafkaQuorumCheck(Reconciliation reconciliation, Admin ac) {
+        this.ac = ac;
+        this.reconciliation = reconciliation;
+        this.quorumInfoFuture = describeMetadataQuorum();
+    }
+
+    /**
+     * Determine whether the given controller can be rolled without affecting the quorum
+     */
+    Future<Boolean> canRoll(int podId) {
```

Comment:
No specific reason here, other than following the existing pattern of having a package scoped method that calls a private method. We can make canRollController() accessible and directly call it instead of canRoll().